plugins/modules/kdeconfig.py

```diff
@@ -0,0 +1,271 @@
+        module.fail_json(msg='Unable to create temporary file', traceback=traceback.format_exc())
+
+    for row in module.params['values']:
+        groups = row['groups']
+        if groups is None:
+            groups = [row['group']]
+        key = row['key']
+        value = row['bool_value']
+        if value is None:
+            value = row['value']
+        run_kwriteconfig(module, kwriteconfig, tmpfile, groups, key, value)
+
+    with open(b_tmpfile, 'rb') as tmpf:
+        b_data = tmpf.read()
+        try:
+            diff['after'] = to_text(b_data)
+        except UnicodeError:
+            diff['after'] = repr(b_data)
+
+    result['changed'] = result['changed'] or diff['after'] != diff['before']
+
+    file_args = module.load_file_common_arguments(module.params)
+
+    if module.check_mode:
+        if not result['changed']:
+            shutil.copystat(b_path, b_tmpfile)
+            uid, gid = module.user_and_group(b_path)
+            shutil.chown(b_tmpfile, uid, gid)
+            if module._diff:
+                diff = {}
+            else:
+                diff = None
+            result['changed'] = module.set_fs_attributes_if_different(file_args, result['changed'], diff=diff)
+        if module._diff:
+            result['diff'] = diff
+        module.exit_json(**result)
+
+    if result['changed']:
+        if module.params['backup'] and os.path.exists(b_path):
+            result['backup_file'] = module.backup_local(result['path'])
+        try:
+            module.atomic_move(b_tmpfile, b_path)
+        except IOError:
+            module.ansible.fail_json(msg='Unable to move temporary file %s to %s, IOError' % (tmpfile, result['path']), traceback=traceback.format_exc())
+
+    if result['changed']:
+        module.set_fs_attributes_if_different(file_args, result['changed'])
+    else:
+        if module._diff:
+            diff = {}
+        else:
+            diff = None
+        result['changed'] = module.set_fs_attributes_if_different(file_args, result['changed'], diff=diff)
+    if module._diff:
+        result['diff'] = diff
+    module.exit_json(**result)
+
+
+def main():
+    single_value_arg = dict(group=dict(type='str'),
+                            groups=dict(type='list', elements='str'),
+                            key=dict(type='str', required=True, no_log=False),
+                            value=dict(type='str'),
+                            bool_value=dict(type='bool'))
+    required_alternatives = [('group', 'groups'), ('value', 'bool_value')]
+    module_args = dict(
+        values=dict(type='list',
+                    elements='dict',
+                    options=single_value_arg,
+                    mutually_exclusive=required_alternatives,
+                    required_one_of=required_alternatives,
+                    required=True),
+        path=dict(type='path', required=True),
+        kwriteconfig_path=dict(type='path'),
+        backup=dict(type='bool', default=False)
```

Comment:
```suggestion
        backup=dict(type='bool', default=False),
```
I would always use a trailling command to avoid unnecessary complex diffs when this is ever extended.